tools/myst/enc/enc.c

```diff
@@ -374,6 +374,13 @@ static uint64_t _vectored_handler(oe_exception_record_t* er)
                 er->context->rsp =
                     *(uint64_t*)(er->context->rsp + IRETFRAME_Rsp);
 
+                /* Always restore signal mask on iretq. For the case of
```

Comment:
Trying to work out if this needs to be done for the linux target.

Reply:
Probably only relevant for dotnet + nested hardware exception.

Reply:
Linux target might handle it already

Reply:
Not sure how though. The host side signal handling code can't restore the signal mask controlled by Mystikos. If iretq isn't intercepted, then I'm not sure where the linux flow will restore the mask. In that case if the same exception happens again, the mask within mystikos would block the signal.

tests/dotnet-ubuntu has a case for multiple exceptions, which seems to be passing for linux target. It would've been interesting to see how p0 and p1 fared on linux target, but unfortunately we run the those suites only on sgx.